hmrc-mongo-common/src/main/scala/uk/gov/hmrc/mongo/MongoUtils.scala

```diff
@@ -27,40 +27,49 @@ trait MongoUtils {
   private val logger: Logger = LoggerFactory.getLogger(classOf[MongoUtils].getName)
 
   def ensureIndexes[A](
-    collection: MongoCollection[A],
-    indexes: Seq[IndexModel],
+    collection    : MongoCollection[A],
+    indexes       : Seq[IndexModel],
     replaceIndexes: Boolean
   )(implicit ec: ExecutionContext
   ): Future[Seq[String]] =
     for {
-      currentIndices <- collection.listIndexes().toFuture().map(_.map(_("name").asString.getValue))
-      res <- Future.traverse(indexes) { index =>
-               collection
-                 .createIndex(index.getKeys, index.getOptions)
-                 .toFuture()
-                 .recoverWith {
-                   case IndexConflict(e) if replaceIndexes =>
-                     logger.warn("Conflicting Mongo index found. This index will be updated")
-                     for {
-                       _      <- collection.dropIndex(index.getOptions.getName).toFuture()
-                       result <- collection.createIndex(index.getKeys, index.getOptions).toFuture()
-                     } yield result
-                 }
-               }
-      indicesToDrop = if (replaceIndexes) currentIndices.toSet.diff(res.toSet + "_id_") else Set.empty
-      _   <- Future.traverse(indicesToDrop) { indexName =>
-               logger.warn(s"Index '$indexName' is not longer defined, removing")
-               collection.dropIndex(indexName).toFuture()
-                 .recoverWith {
-                   // could be caused by race conditions between server instances
-                   case IndexNotFound(e) => Future.successful(())
-                 }
-             }
+      currentIndices   <- collection.listIndexes().toFuture().map(_.map(_("name").asString.getValue))
+      requestedIndices =  indexes.map(_.getOptions.getName)
+      orphanedIndices  =  currentIndices.toSet.diff(requestedIndices.toSet + "_id_")
+      _                <- if (orphanedIndices.nonEmpty)
+                            if (replaceIndexes)
+                              Future.traverse(orphanedIndices) { indexName =>
+                                logger.warn(s"Index '$indexName' is not longer defined for ${collection.namespace}, removing")
+                                collection.dropIndex(indexName).toFuture()
+                                  .recoverWith {
+                                    // could be caused by race conditions between server instances
+                                    case IndexNotFound(e) => Future.unit
+                                  }
+                              }
+                            else
+                              Future.successful(logger.warn(s"The following indices exist in mongo for ${collection.namespace}, but are (no longer) provided to ensureIndexes: ${orphanedIndices.mkString(", ")}"))
```

Comment:
Although indices and indexes are both correct as the plural for index, as mongo has already fixed on indexes - and we have used indexes too (like in ensureIndexes), I feel we should just use indexes through out. I would replace all occurrences of indices with indexes.